### PR TITLE
Frontend scaffolding (prettier, navbar, testing)

### PR DESCRIPTION
**Pull request by allisonking**
Some progress towards https://github.com/ethyca/fides/issues/676. Wanted to put in a PR before this got too big, since it was already touching a lot of files.

### Code Changes

* [x] Add import sorting similar to fidesops https://github.com/ethyca/fidesops/pull/514
* [x] Add baseUrl for easier imports
* [x] Add prettier
* [x] Run prettier
* [x] Refactor Nav bar into a Layout component to make it easier to duplicate across pages
* [x] Add testing to nav bar

### Steps to Confirm

* [x] Run `npm run dev` and poke around!

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

Not really any big UI changes, but swapped out the fidesops navbar for one that looks more like the one in Balsamiq
image

Also added a page for `/dataset` but it's empty right now.
image

(that active state for "Datasets" in the nav bar probably isn't what we want it to look like, but I'm not gonna mess with CSS before we have an actual design) 

After this, we should be ready to actually display some data on the /dataset route

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,71 +6,72 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/)
 
 The types of changes are:
 
-* `Added` for new features.
```
Comment (ThomasLaPiana) [inline CHANGELOG.md:9]: that MD lint though 😆

Comment (allisonking) [reply CHANGELOG.md:9]: ugh oops didn't see that, I can undo it?

Comment (allisonking) [reply CHANGELOG.md]: I decided to put the CHANGELOG.md in the .prettierignore

Comment (ThomasLaPiana) [reply CHANGELOG.md]: oh it didn't actually bother me, I just assumed you had a VS code linter for yaml or something lol

**Review by elliotbonneville — CHANGES_REQUESTED** [review]
This looks great, @allisonking. Love the cleanup work and improvements to tooling, etc. throughout. `NavLink` is also a well-implemented component. Thanks for cleaning that up.

Just one comment here around a use of `any` – if you can find away around using it I'm happy to approve!

File: clients/admin-ui/src/features/common/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      >
+        {title}
+      </Button>
+    </NextLink>
+  );
+};
+
+const NavBar = () => {
+  const { data: session } = useSession();
+  const username: string | any = session?.username;
```
[the remark this answers is not in the document]
Comment (allisonking) [reply clients/admin-ui/src/features/common/NavBar.tsx]: mm okay yeah looks like this was copypasta'd, so we'll need to fix on fidesops too. I'm not sure what behavior we want when there isn't a username, so I [made it just go to an empty string for now](https://github.com/ethyca/fides/pull/679/commits/2bce84deabf35fc8ac93b7f9445fe112b49c8c43). It sounds like our auth will change pretty soon, so don't want to build anything complex around this until we have that in place.

[the remark this answers is not in the document]
Comment (elliotbonneville) [reply clients/admin-ui/src/features/common/NavBar.tsx]: Perfect, thanks for that. Auth will impact that so good call!

**Review by elliotbonneville — APPROVED** [review]

**Pull request merged**